### PR TITLE
Fixes the compile_commands.json recipe

### LINKED ISSUE #9895: make compile_commands.json fix
### Description of the enhancement or error report
The command `make compile_commands.json` may fail when a large number of source files are present. This was observed when all the modules were included in an `MooseApp`'s build.

### Rationale for the enhancement or information for reproducing the error
A fix can be applied that fixes this issue in `framework/moose.mk`, but it requires Make v4+
Another workaround is to customize an app's Makefile to include only the needed modules

### Identified impact
Immediate fix for users with Make v4+. No impact for other users.

### PR DESCRIPTION
**Pull request by BruceWyxv**
closes #9895

only modifies `framework/moose.mk`

### CODE REVIEW

**Review by dschwen — APPROVED** [review]
Looks good to me

**Comment by BruceWyxv** [comment]
I see that the check `App tests` failed.

I'm not privileged enough to see the details so I don't know how to fix it. Is this something I need to worry about?

**Review by permcody — APPROVED** [review]

File: framework/moose.mk [hunk 1]
```diff
@@ -219,10 +219,17 @@ clobberall:: clobber
 
 compile_commands_all_srcfiles := $(moose_srcfiles) $(srcfiles)
 compile_commands.json::
-	@echo $(CURDIR) > .compile_commands.json
-	@echo $(libmesh_CXX) >> .compile_commands.json
-	@echo $(libmesh_CPPFLAGS) $(libmesh_CXXFLAGS) $(app_INCLUDES) $(libmesh_INCLUDE) $(ADDITIONAL_INCLUDES) >> .compile_commands.json
-	@echo $(compile_commands_all_srcfiles) >> .compile_commands.json
+	ifeq (4.0,$(firstword $(sort $(MAKE_VERSION) 4.0)))
```
Comment (BruceWyxv) [inline framework/moose.mk]: I just discovered that there is whitespace before `ifeq`, `else`, and `endif` in this commit. As a result, Make incorrectly interprets these as bash directives and the whole process fails. What is the best way to *>ahem<* patch this patch?

**Comment by permcody** [comment]
Just make another PR - you can reference the existing ticket though. You don't necessarily need to open new tickets for things like this. It happens.

**Pull request merged**